src/core/job.c

```diff
@@ -1051,6 +1051,8 @@ int job_finish_and_invalidate(Job *j, JobResult result, bool recursive, bool alr
                         job_add_to_gc_queue(other->job);
                 }
 
+        (void) unit_submit_to_start_when_upheld_queue(NULL, 0, u);
```

Comment:
this should be a separate commit, no? also, please add a brief comment why this is useful here

Reply:
hmm, this shouldn't add `u` to the queue, but all units that come after it. it should be moved inside the UNIT_ATOM_BEFORE loop immediately preceeding here, and operate on `other`